community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/Status.java

```diff
@@ -400,7 +400,7 @@ public int hashCode()
     public enum Classification
     {
         /** The Client sent a bad request - changing the request might yield a successful outcome. */
-        ClientError( TransactionEffect.NONE,
+        ClientError( TransactionEffect.ROLLBACK,
```

Comment:
I'm probably missing context here - but could you expand on why we want to change this? 

The original intention for not rolling back the transaction on client errors was that client errors would be something like syntax errors, and that if you're using an interactive shell or some such it'd be really frustrating if a big transaction gets rolled back because you make a spelling mistake or some similar error.